implementations/rust/ockam/ockam_node/src/router/utils.rs

```
@@ -0,0 +1,74 @@
+use super::{AddressRecord, NodeState, Router};
+use crate::tokio::sync::mpsc::Sender;
+use crate::{
+    error::Error,
+    relay::{RelayMessage, PROC_ADDR_SUFFIX},
+    NodeReply, NodeReplyResult, Reason,
+};
+
+use ockam_core::{Address, AddressSet, Result};
+
+/// Receive an address and resolve it to a sender
+///
+/// This function only applies to local address types, and will
+/// fail to resolve a correct address if it given a remote
+/// address.
+pub(super) async fn resolve(
+    router: &mut Router,
+    addr: &Address,
+    reply: &Sender<NodeReplyResult>,
+    wrap: bool,
+) -> Result<()> {
+    trace!("Resolving worker address '{}'", addr);
+
+    let primary_address;
+    if let Some(p) = router.map.addr_map.get(addr) {
+        primary_address = p.clone();
+    } else {
+        reply
+            .send(NodeReply::no_such_worker(addr.clone()))
+            .await
+            .map_err(|_| Error::InternalIOFailure)?;
+
+        return Ok(());
+    }
+
+    match router.map.internal.get(&primary_address) {
+        Some(record) => reply.send(NodeReply::sender(addr.clone(), record.sender(), wrap)),
+        None => reply.send(NodeReply::no_such_worker(addr.clone())),
+    }
+    .await
+    .expect("Ockam node internal I/O failed!");
```

Should it fail silently or return error instead?